src/UserAuth.php

```diff
@@ -26,12 +26,8 @@ public function __construct(CurrentUser $currentUser, ResponseFactoryInterface $
         $this->responseFactory = $responseFactory;
     }
 
-    public function authenticate(ServerRequestInterface $request): ?IdentityInterface
+    public function authenticate(ServerRequestInterface $request): IdentityInterface
     {
-        if ($this->currentUser->isGuest()) {
-            return null;
-        }
-
```

Comment:
It's break logic of Yii Auth package in `Authentication` middleware: https://github.com/yiisoft/auth/blob/27a4e9ac4e89d810625c73f1578586d2ecdc9c27/src/Middleware/Authentication.php#L48C43-L48C53